### PR TITLE
Added api index

### PR DESCRIPTION
**Pull request by rominariedel**
# Summary

- Articles are listed paginated via api

# Trello Cards

https://trello.com/c/gxoypfn1/23-crear-lista-de-articulos-usando-api-rest

### CODE REVIEW

File: app/assets/javascripts/articles.js [hunk 1]
```diff
@@ -35,3 +35,74 @@ $(document.body).on('click', 'p.save-article', function (e) {
...
+    url: url,
+    type: 'GET',
+    success: function(data){
+      resetListedArticlesStatus(data.articles);
+      resetEditArticlesStatus(data.articles);
+    }
+  })
+})
+
+function resetListedArticlesStatus(articles){
```
Comment (epintos) [inline app/assets/javascripts/articles.js]: Just for you to know. In JS is better to define the functions before the invocation. So this functions should go above

File: app/assets/javascripts/articles.js [hunk 2]
```diff
@@ -35,3 +35,74 @@ $(document.body).on('click', 'p.save-article', function (e) {
...
+
+  var articlesDiff = $(document).find(selector).length - articles.length;
+  
+  if(articlesDiff > 0) {
+    removeArticles(articles.length, articlesDiff, selector);
+  }
+}
+
+function replaceDataId(elem, article) {
+  elem.attr('data-id', article.id);
```
Comment (epintos) [inline app/assets/javascripts/articles.js]: Hmm. cant you do `.data('id')` here?

File: app/assets/javascripts/articles.js [hunk 3]
```diff
@@ -35,3 +35,74 @@ $(document.body).on('click', 'p.save-article', function (e) {
...
+  elem.find('[data-id]').attr('data-id', article.id);
+}
+
+function replaceTitle(elem, article) {
+  elem.find('div.title > a').html(article.title);
+  elem.find('input.title-input').val(article.title);
+}
+
+function replaceReferences(elem, article) {
+  elem.find('[href]').not('[href $= "edit" ]').attr('href', "/articles/" + article.id);
```
Comment (epintos) [inline app/assets/javascripts/articles.js]: The url can be returned by the api

File: app/models/article.rb [hunk 4]
```diff
@@ -11,6 +11,8 @@
 #
 
 class Article < ActiveRecord::Base
+  paginates_per 2
```
Comment (epintos) [inline app/models/article.rb]: why 2?

Comment (rominariedel) [inline app/models/article.rb]: It's the number I thought it fit better in the page, should I be using another per page?

File: app/controllers/api/v1/articles_controller.rb [hunk 5]
```diff
@@ -20,4 +22,14 @@ def update
   def article_params
     params.require(:article).permit(:title, :text)
   end
+
+  def set_articles_path
```
Comment (epintos) [inline app/controllers/api/v1/articles_controller.rb]: You can use ActiveModelSerializers for this or it didnt work?

**Pull request closed without merging**